flow/IThreadPoolTest.actor.cpp

```diff
@@ -60,6 +61,89 @@ TEST_CASE("/flow/IThreadPool/NamedThread") {
 	return Void();
 }
 
+struct ThreadSafePromiseStreamSender final : IThreadPoolReceiver {
+	ThreadSafePromiseStreamSender(
+		ThreadReturnPromiseStream<std::string>* notifications)
+		 : notifications(notifications) {}
+	void init() override {}
+
+	struct GetNameAction final : TypedAction<ThreadSafePromiseStreamSender, GetNameAction> {
+		double getTimeEstimate() const override { return 3.; }
+	};
+
+	void action(GetNameAction& a) {
+		pthread_t t = pthread_self();
+		const size_t arrayLen = 16;
+		char name[arrayLen];
+		int err = pthread_getname_np(t, name, arrayLen);
+		if (err != 0) {
+			std::cout << "Get name failed with error code: " << err << std::endl;
+			notifications->sendError(platform_error());
+			return;
+		}
+		notifications->send(std::move(name));
```

Comment:
nit: this `std::move` isn't doing anything. Since `name` is a char array, in both cases, this is delegating to the `std::string(const char*)` constructor. Moving a pointer is the same as copying it.

Reply:
Nice catch, removed std::move